runtime/str.go

```diff
@@ -753,6 +753,12 @@ func strInterpolate(f *Frame, format string, values *Tuple) (*Object, *BaseExcep
 					return nil, raised
 				}
 				val = s.Value()
+			} else if matches[7] == "o" {
+				if o.isInstance(LongType) {
+					val = toLongUnsafe(o).Value().Text(8)
```

Comment:
Huh, that is very weird. It appears that CPython does not call `__hex__`/`__oct__`. So I'm confused about what the code in stringobject.c is actually doing.